calculation/calculation.go

```diff
@@ -495,50 +510,55 @@ func TruncateEmissions(program types.Program, emissionsByPool map[string]uint64)
 
 // Compute the total LP token days that each owner has; We multiply the LP tokens by seconds they were locked, and then divide by 86400.
 // This effectively divides the LP tokens by the fraction of the day they are locked, to prevent someone locking in the last minute of the day to receive rewards
-func TotalLPDaysByOwnerAndAsset(positions []types.Position, poolLookup PoolLookup, minSlot uint64, maxSlot uint64) (map[string]map[chainsync.AssetID]uint64, map[chainsync.AssetID]uint64) {
-	lpDaysByOwner := map[string]map[chainsync.AssetID]uint64{}
-	lpDaysByAsset := map[chainsync.AssetID]uint64{}
+func TotalLPDaysByOwnerAndAsset(positions []types.Position, poolLookup PoolLookup, minSlot uint64, maxSlot uint64) (map[string]map[shared.AssetID]uint64, map[shared.AssetID]uint64) {
+	lpDaysByOwner := map[string]map[shared.AssetID]uint64{}
+	lpDaysByAsset := map[shared.AssetID]uint64{}
 	for _, p := range positions {
-		for assetId, amount := range p.Value.Assets {
-			if poolLookup.IsLPToken(assetId) {
-				// Compute the (truncated) start and end time,
-				startTime := p.Slot
-				if startTime < minSlot {
-					startTime = minSlot
-				}
-				endTime := p.SpentSlot
-				if p.SpentTransaction == "" || p.SpentSlot > maxSlot {
-					endTime = maxSlot
-				}
-				if endTime == startTime {
-					continue
-				}
-				// so we can compute what fraction of the day this position counts for
-				secondsLocked := endTime - startTime
+		for policy, policyMap := range p.Value {
+			for _, amount := range policyMap {
 
-				weight := big.NewInt(0).SetUint64(secondsLocked)
-				weight = weight.Mul(weight, amount.BigInt())
-				weight = weight.Div(weight, big.NewInt(0).SetUint64(maxSlot-minSlot))
+				assetId := shared.AssetID(policy)
```

Comment:
the assetId is not the policy ID. this should be `shared.FromSeparate(policy, assetName)`, where `assetName` is the `_` in the nested for above